### ISSUE TITLE
Compile problem on OS X 10.9.5

### ISSUE THREAD

**Issue by oscii** [EIS-Bonn/Luzzu#7]
Hi folks!
I can't compile Luzzu of OS X 10.9.5. Unfortunately, I can't post the full output of the `mvn clean install -X` command here (because it is too long), so I post just an exception message and a stderr.

```
[DEBUG] Source roots:
[DEBUG]  /Users/oscii/Activities/Dev/ailab/RussianMuseum/Luzzu/luzzu-semantics/src/main/java
[DEBUG] Command line options:
[DEBUG] -d /Users/oscii/Activities/Dev/ailab/RussianMuseum/Luzzu/luzzu-semantics/target/classes -classpath /Users/oscii/Activities/Dev/ailab/RussianMuseum/Luzzu/luzzu-semantics/target/classes:/Users/oscii/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/Users/oscii/.m2/repository/org/mockito/mockito-core/1.9.5/mockito-core-1.9.5.jar:/Users/oscii/.m2/repository/org/objenesis/objenesis/1.0/objenesis-1.0.jar:/Users/oscii/.m2/repository/org/hamcrest/hamcrest-all/1.3/hamcrest-all-1.3.jar:/Users/oscii/.m2/repository/org/apache/jena/jena-tdb/1.0.2/jena-tdb-1.0.2.jar:/Users/oscii/.m2/repository/org/apache/jena/jena-arq/2.11.2/jena-arq-2.11.2.jar:/Users/oscii/.m2/repository/org/apache/httpcomponents/httpclient/4.2.6/httpclient-4.2.6.jar:/Users/oscii/.m2/repository/org/apache/httpcomponents/httpcore/4.2.5/httpcore-4.2.5.jar:/Users/oscii/.m2/repository/commons-codec/commons-codec/1.6/commons-codec-1.6.jar:/Users/oscii/.m2/repository/com/github/jsonld-java/jsonld-java/0.4/jsonld-java-0.4.jar:/Users/oscii/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.3.3/jackson-core-2.3.3.jar:/Users/oscii/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.3.3/jackson-databind-2.3.3.jar:/Users/oscii/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.3.0/jackson-annotations-2.3.0.jar:/Users/oscii/.m2/repository/org/apache/httpcomponents/httpclient-cache/4.2.6/httpclient-cache-4.2.6.jar:/Users/oscii/.m2/repository/org/apache/jena/jena-core/2.11.2/jena-core-2.11.2.jar:/Users/oscii/.m2/repository/org/apache/jena/jena-iri/1.0.2/jena-iri-1.0.2.jar:/Users/oscii/.m2/repository/xerces/xercesImpl/2.11.0/xercesImpl-2.11.0.jar:/Users/oscii/.m2/repository/xml-apis/xml-apis/1.4.01/xml-apis-1.4.01.jar:/Users/oscii/.m2/repository/log4j/log4j/1.2.16/log4j-1.2.16.jar:/Users/oscii/.m2/repository/org/slf4j/slf4j-api/1.7.5/slf4j-api-1.7.5.jar:/Users/oscii/.m2/repository/org/slf4j/jcl-over-slf4j/1.7.5/jcl-over-slf4j-1.7.5.jar:/Users/oscii/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar:/Users/oscii/.m2/repository/org/slf4j/slf4j-jdk14/1.7.5/slf4j-jdk14-1.7.5.jar:/Users/oscii/.m2/repository/org/apache/commons/commons-collections4/4.0/commons-collections4-4.0.jar: -sourcepath /Users/oscii/Activities/Dev/ailab/RussianMuseum/Luzzu/luzzu-semantics/src/main/java: -s /Users/oscii/Activities/Dev/ailab/RussianMuseum/Luzzu/luzzu-semantics/target/generated-sources/annotations -g -nowarn -target 1.7 -source 1.7 -encoding UTF-8
[DEBUG] incrementalBuildHelper#beforeRebuildExecution
[INFO] Compiling 11 source files to /Users/oscii/Activities/Dev/ailab/RussianMuseum/Luzzu/luzzu-semantics/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Luzzu .............................................. SUCCESS [  0.410 s]
[INFO] Luzzu Semantics Module ............................. FAILURE [  5.157 s]
[INFO] Luzzu Assessment Module ............................ SKIPPED
[INFO] Luzzu Operations Module ............................ SKIPPED
[INFO] Luzzu Annotations Module ........................... SKIPPED
[INFO] Luzzu Quality Metric Language ...................... SKIPPED
[INFO] Luzzu IO Module .................................... SKIPPED
[INFO] Luzzu Communications Module ........................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.763 s
[INFO] Finished at: 2015-01-08T00:51:45+03:00
[INFO] Final Memory: 27M/81M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project luzzu-semantics: Fatal error compiling: invalid target release: 1.7 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project luzzu-semantics: Fatal error compiling
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Fatal error compiling
	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:796)
	at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:129)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: org.codehaus.plexus.compiler.CompilerException: invalid target release: 1.7
	at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:191)
	at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:169)
	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:785)
	... 22 more
Caused by: java.lang.IllegalArgumentException: invalid target release: 1.7
	at com.sun.tools.javac.main.RecognizedOptions$GrumpyHelper.error(RecognizedOptions.java:75)
	at com.sun.tools.javac.main.RecognizedOptions$14.process(RecognizedOptions.java:380)
	at com.sun.tools.javac.api.JavacTool.processOptions(JavacTool.java:247)
	at com.sun.tools.javac.api.JavacTool.getTask(JavacTool.java:207)
	at com.sun.tools.javac.api.JavacTool.getTask(JavacTool.java:53)
	at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:115)
	... 24 more
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :luzzu-semantics
```

```
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-jdk14/1.7.5/slf4j-jdk14-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
Schemagen failed to run:
Failed to read input source file:src/main/resources/vocabularies/daq/daq.trig
Caused by: Not found: file:src/main/resources/vocabularies/daq/daq.trig
     [java] Java Result: 1
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-jdk14/1.7.5/slf4j-jdk14-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
Schemagen failed to run:
Failed to read input source file:src/main/resources/vocabularies/void/void.ttl
Caused by: Not found: file:src/main/resources/vocabularies/void/void.ttl
     [java] Java Result: 1
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-jdk14/1.7.5/slf4j-jdk14-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
Schemagen failed to run:
Failed to read input source file:src/main/resources/vocabularies/qr/qr.trig
Caused by: Not found: file:src/main/resources/vocabularies/qr/qr.trig
     [java] Java Result: 1
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-jdk14/1.7.5/slf4j-jdk14-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
Schemagen failed to run:
Failed to read input source file:src/main/resources/vocabularies/cube/cube.ttl
Caused by: Not found: file:src/main/resources/vocabularies/cube/cube.ttl
     [java] Java Result: 1
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-jdk14/1.7.5/slf4j-jdk14-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
Schemagen failed to run:
Failed to read input source file:src/main/resources/vocabularies/cube/sdmx-attribute.ttl
Caused by: Not found: file:src/main/resources/vocabularies/cube/sdmx-attribute.ttl
     [java] Java Result: 1
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/oscii/.m2/repository/org/slf4j/slf4j-jdk14/1.7.5/slf4j-jdk14-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
Schemagen failed to run:
Failed to read input source file:src/main/resources/vocabularies/lmi/lmi.trig
Caused by: Not found: file:src/main/resources/vocabularies/lmi/lmi.trig
     [java] Java Result: 1
```

Any suggestions?

**Comment by s-londono** [comment]
Dear Eugene,

This error means that your mvn script is using a JDK not compatible with 
version 1.7.
Firstly, please verify the default java version installed in your 
system, by running the command:
java -version

If your Java version is 1.7 or above, please check your mvn script and 
make sure that the JAVA_HOME environment variable is not being 
ovewritten somehow (for details, please refer to: 
http://stackoverflow.com/questions/14043042/compiling-java-7-code-via-maven)

Kind regards,

The Luzzu team.

**Comment by oscii** [comment]
Thanks!

I removed all hardcoded $JAVA_HOME stuff from maven script and set this environment variable in my .zshrc.

I managed to build Luzzu only with tests skipped. I had a weird log on failed test: https://gist.github.com/oscii/13d727e3d79388433da6

But webapp works normally so far.

**Comment by jerdeb** [comment]
Hi Eugene,

Can you please tell us your java version please?

Cheers,
Jeremy

**Comment by oscii** [comment]
Hi Jeremy,

Sure:

```bash
~ $ java -version
java version "1.7.0_71"
Java(TM) SE Runtime Environment (build 1.7.0_71-b14)
Java HotSpot(TM) 64-Bit Server VM (build 24.71-b01, mixed mode)
~ $ echo $JAVA_HOME
/Library/Java/JavaVirtualMachines/jdk1.7.0_71.jdk/Contents/Home
```

Best,
Eugene

**Comment by jerdeb** [comment]
Hi Eugene,

It seems that there was a problem with powermock and java version 1.7.0_71. It will be fixed.

Cheers,
Jeremy

**Comment by s-londono** [comment]
The errors occurring during the test phase are caused by changes introduced in new versions of the JVM, which conflict with some powermock functionalities. In Java 7, These problems are solved setting the JVM parameter -XX:-UseSplitVerifier, which was done by adding the appropriate surefire plugin configuration to Luzzu's POM.

Fixed on commit: 1ec71e943c89fce4ce988de8104759860b9669ae
